### PR TITLE
Allow enabling metrics/logging without teardown

### PR DESCRIPTION
**Pull request by patricklucas**
By always activating the Prometheus metrics reporter and by using `helm
upgrade --install`, the setup script can be run multiple times
idempotently to enable logging and/or metrics without first tearing down
the existing installation.

Another side-effect of using `helm upgrade --install` is that we don't actually have to do Helm version discovery in `setup.sh` anymore.

### CODE REVIEW

File: values-elasticsearch.yaml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+
 resources:
   requests:
     cpu: "100m"
-    memory: "2Gi"
+    memory: "1Gi"
   limits:
     cpu: "1000m"
-    memory: "2Gi"
+    memory: "1Gi"
```
Comment (NicoK) [inline values-elasticsearch.yaml:12]: I was actually playing around with this as well and wanted to verify a bit more before suggesting.
Just FYI: reducing to 500Mi (and heap: 256Mi) caused unexpected restarts. The 1Gi setting you propose here seems stable for now

File: values-kibana.yaml [hunk 2]
```diff
@@ -7,10 +7,10 @@ replicas: 1
 resources:
   requests:
     cpu: "100m"
-    memory: "500Mi"
+    memory: "256Mi"
   limits:
     cpu: "1000m"
-    memory: "1Gi"
+    memory: "256Mi"
```
Comment (NicoK) [inline values-kibana.yaml:13]: Why do you think we need to reduce the limit as well?

Comment (NicoK) [reply values-kibana.yaml:13]: I did not test these reductions and suppose you verified those are working even for larger sets of logs?

**Comment by patricklucas** [comment]
- `helm install --upgrade` in the PR description was a typo, I'll fix it.
- I used the same memory configs we used to use for the demo components built into the chart; of course there are situations and levels of usage that could require a user to increase these values but I never actually hit that when working with the demo components with these limits. I think starting low (but stable with 1 or 2 jobs) is a perfectly fine place to start, and if we start seeing instability, perhaps as the E/K charts update (we don't pin a version), we can increase them.
- I can do a little bit of commit reorg, but I don't think I would make the `upgrade --install` and values file changes separate as that would reduce the "atomicity" of the commit. The reason for the whole change is those two things together and it's clear from the commit message what's going on.

**Comment by patricklucas** [comment]
I split the first commit into:

```
commit 4c0b0e21d61d184f5a7a8f466b628f9358fb5f23

    Minor chart values optimizations
    
    - Allow Elasticsearch to run as a non-privileged container by disabling
      its requirement for sysctl vm.max_map_count >= 262144
    - Specify an (unused) admin password for Grafana so that it does not
      regenerate its Kubernetes Secret on every upgrade which causes a
      needless restart

commit 35565726f028710ad993da1964d6217121e94f50

    Minor shell optimizations
    
    Remove a function param to make calls a bit more explicit and remove
    some code duplication.

commit ea5a4019102831c9795a140e333204b30e035f56

    Allow enabling metrics/logging without teardown
    
    By always activating the Prometheus metrics reporter and by using `helm
    install --upgrade`, the setup script can be run multiple times
    idempotently to enable logging and/or metrics without first tearing down
    the existing installation.
```

**Pull request merged**